src/charm.py

```diff
@@ -122,51 +177,56 @@ def mysql(self) -> MySQL:
         """Returns MySQL object"""
         mysql_config = {
             "app_name": self.model.app.name,
-            "host": self.hostname,
+            "host": self.unit_ip,
             "port": self.model.config["port"],
             "user_name": "root",
-            "mysql_root_password": self._stored.mysql_setup[
-                "MYSQL_ROOT_PASSWORD"
-            ],
+            "mysql_root_password": self.mysql_root_password,
         }
         return MySQL(mysql_config)
 
     @property
-    def hostname(self) -> str:
-        """Unit hostname"""
-        unit_id = self.unit.name.split("/")[1]
-        return "{0}-{1}.{0}-endpoints".format(self.model.app.name, unit_id)
+    def unit_ip(self) -> str:
+        """Returns unit's IP"""
+        return str(self.model.get_binding(PEER).network.bind_address)
```

Comment:
Note that `bind_address` may return `None` on startup.
For this reason I would be careful with converting to string ('None' as string evaluates as True!).
I have this:
```python
    @property
    def private_address(self) -> Optional[IPv4Address]:
        return self.model.get_binding(PEER).network.bind_address
```